r/R/dataset-format.R

```diff
@@ -309,14 +309,52 @@ FileWriteOptions <- R6Class("FileWriteOptions",
   inherit = ArrowObject,
   public = list(
     update = function(table, ...) {
+      compose_err_header <- function(unsupported_passed_args) {
+        paste0(
+          oxford_paste(unsupported_passed_args),
+          ngettext(length(unsupported_passed_args),
+                   " is not a valid argument ",
+                   " are not valid arguments "),
+          "for your chosen `format`."
+        )
+      }
```

Comment:
Nice, pulling this out into a separate function like this. I wonder if we could expand this even more to include the actual `abort()` call, and the following:

```
unsupported_passed_args <- setdiff(names(args), supported_args)

 if (length(unsupported_passed_args)) {
   err_header <- compose_err_header(unsupported_passed_args)
   abort(err_header)
 }
```

Basically, something like:

```
compose_err_header <- function(passed_args, format) {
   ...
   if (length(...) > 0) { 
      abort(...)
   }
}
```

or:
```
compose_err_header <- function(passed_args, supported_args) {
   ...
   if (length(...) > 0) { 
      abort(...)
   }
}
```

Reply:
This is slightly more complicated if we want to support type-specific error bullet points. Normally, I wouldn't be in favour as to me, this function will then try to do too much. I'll give it a go and see what it looks like.